api/nodes/serializers.py

```diff
@@ -198,8 +198,12 @@ class Meta:
         type_ = 'styled-citations'
 
 def get_license_details(node, validated_data):
-    license = node.license if isinstance(node, Preprint) else node.node_license
-
+    if node:
+        license = node.license if isinstance(node, Preprint) else node.node_license
+    else:
+        license = None
+    if ('license_type' not in validated_data and not (license and license.node_license.license_id)):
+        raise exceptions.ValidationError(detail='License ID must be provided for a Node License.')
```

Comment:
This will modify updating license behavior as well, so we should probably test that.